### PR TITLE
Add userData to EKS nodes

### PR DESCRIPTION
**Pull request by daxmc99**
Problem: We want to be able to specify the user Data field on nodes
created for EKS.

Solution: Change cloudformation template yaml to allow for customUserData
field. Create 'userData' api field in UI.

Issue https://github.com/rancher/rancher/issues/16300
UI Change: https://github.com/rancher/rancher/issues/17711

### CODE REVIEW

**Comment by daxmc99** [comment]
@nathan-jenan-rancher

File: drivers/eks/eks_driver.go [hunk 1]
```diff
@@ -194,6 +199,16 @@ func (d *Driver) GetDriverCreateOptions(ctx context.Context) (*types.DriverFlags
...
 		},
 	}
+	// If this is used it will overwrite the default values in the template and use what is given here
+	// Expectation: leave this as "" otherwise show what was there previously with a warning to be careful
+	//	modifying it
+	driverFlag.Options["user-data"] = &types.Flag{
+		Type:  types.StringType,
+		Usage: "Pass user-data to the nodes to perform automated configuration tasks",
+		Default: &types.Default{
+			DefaultString: "",
```
Comment (nathan-jenan-rancher) [inline drivers/eks/eks_driver.go]: Strings should default to`""` (empty string) anyway, right?  I think this is unnecessary unless there is a compelling reason I'm not seeing.

File: drivers/eks/eks_driver.go [hunk 2]
```diff
@@ -291,6 +309,8 @@ func (state *state) validate() error {
 		return fmt.Errorf("if AssociateWorkerNodePublicIP is set to false a VPC and subnets must also be provided")
 	}
 
+	//TODO Add userData validation
```
Comment (nathan-jenan-rancher) [inline drivers/eks/eks_driver.go]: Either implement validation or remove TODO

File: drivers/eks/eks_driver.go (hunk 1 above)
Comment (daxmc99) [reply drivers/eks/eks_driver.go]: Unfortunately not, it is set to nil if it is not initialized which becomes an issue in testing where GET requests return cluster objects with a nil subnet.

**Comment by nathan-jenan-rancher** [comment]
LGTM

**Comment by nathan-jenan-rancher** [comment]
LGTM

**Comment by nathan-jenan-rancher** [comment]
LGTM

**Comment by cjellick** [comment]
LGTM

**Pull request merged**